test/unit/tool_util/test_wait.py

```diff
@@ -9,6 +11,7 @@ def __init__(self):
         self.sleeps = []
 
     def sleep(self, delta):
+        sleep(delta)
```

Comment:
I don't think we need to actually do time.sleep here

Reply:
What would be the alternative?

Reply:
This seems to be last open point here. Could you give me a hint on how to avoid `time.sleep`?

Reply:
The previous implementation of this method was much more testable. I don't maybe understand why you changed the logic. I guess now you'd need to send in an override of sleep and time and that probably isn't worth it?

Reply:
> I don't maybe understand why you changed the logic.

I noted that the time spent for tests that timeout (controlled via planemo's `--test_timeout` parameter) is a multiple of the time that is specified. If I remember correctly it was off by at least a factor of 2. The reason is that the time needed to execute `function` (repeatedly) is not count in `test_timeout`. 

I found this slightly  annoying. But I completely agree that this makes it harder/impossible(?) to test. 

An alternative might be a reformulation of the help text of planemo's parameter.

If you like you can check the effect with the following "program"

```
import time

class TimeoutAssertionError(AssertionError):
    """Derivative of AssertionError indicating wait_on exceeded max time."""

    def __init__(self, message):
        super().__init__(message)


def wait_on(
    function,
    desc,
    timeout,
    delta=0.25,
    polling_backoff = 0,
    sleep_ = None,
):
    """Wait for function to return non-None value.
    Grow the polling interval (initially ``delta`` defaulting to 0.25 seconds)
    incrementally by the supplied ``polling_backoff`` (defaulting to 0).
    Throw a TimeoutAssertionError if the supplied timeout is reached without
    supplied function ever returning a non-None value.
    """
    sleep = sleep_ or time.sleep
    total_wait = 0.0
    start = time.time()
    while True:
        print(f"{time.time() - start} {total_wait}")
        if total_wait > timeout:
            raise TimeoutAssertionError(TIMEOUT_MESSAGE_TEMPLATE.format(total_wait, desc))
        value = function()
        if value is not None:
            return value
        total_wait += delta
        sleep(delta)
        delta += polling_backoff


def work():
    time.sleep(0.1)

wait_on(work, "blah", 900)
```